File: 021-Merge-Two-Sorted-Lists/Solution 2.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        ret = ListNode(-1)
        ptr = ret
        while l1 is not None and l2 is not None:
            if l1.val <= l2.val:
                ptr.next, l1 = l1, l1.next
                ptr = ptr.next
            else:
                ptr.next, l2 = l2, l2.next
                ptr = ptr.next
            
        if l1 is not None:
            ptr.next = l1
        if l2 is not None:
            ptr.next = l2
        return ret.next

File: Solutions/021-Merge-Two-Sorted-Lists/Solution 2.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode  l3Head(-1);
        ListNode * l3 = &l3Head;
        while(l1 != NULL && l2 != NULL){
            if( l1->val <= l2->val){
                l3->next = l1;
                l1 = l1->next;
                l3 = l3->next;
                l3->next = NULL;
            }else {
                l3->next = l2;
                l2 = l2->next;
                l3 = l3->next;
                l3->next = NULL;
            }
        }
        if (l1 == NULL){
            l3->next = l2;
        }else if(l2 == NULL){
            l3->next = l1;
        } 
            
        return l3Head.next;
    }
};
